File: 백준/Gold/1806. 부분합/부분합.cc
#include<iostream>
using namespace std;

int arr[100005];
int n, s, tmp, st, en, ans = 100005;

int main()
{
	ios::sync_with_stdio(0); cin.tie(0);
	cin >> n >> s;
	for (int i = 0; i < n; i++) cin >> arr[i];

	while (st <= n && en <= n) {
		if (tmp < s) tmp += arr[en++];
		else {
			ans = min(ans, en - st);
			tmp -= arr[st++];
		}
	}
	if (ans == 100005) cout << 0;
	else cout << ans;
	return 0;
}

File: 백준/Gold/1806. 부분합/부분합.java
import java.io.*;
import java.util.*;

class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int n = Integer.parseInt(st.nextToken());
		int s = Integer.parseInt(st.nextToken());
		int[] arr = new int[n];
		st = new StringTokenizer(br.readLine());
		for(int i = 0; i < n; i++) arr[i] = Integer.parseInt(st.nextToken());
		int left = 0, right = 0, sum = 0, ans = n+1;
		
		while(left < n){
			if(sum < s){
				if(right == n) break;
				sum += arr[right++];
			}else{
				ans = Math.min(ans, right - left);
				sum -= arr[left++];
			}
		}
		System.out.println(ans == n+1? 0 : ans);
	}
}